packages/cli/src/commands/rewards/show.ts

```diff
@@ -0,0 +1,261 @@
+import { cli } from 'cli-ux'
+import { EventLog } from 'web3/types'
+import { BaseCommand } from '../../base'
+import { newCheckBuilder } from '../../utils/checks'
+import { Flags } from '../../utils/command'
+
+interface VoterReward {
+  name: string
+  group: string
+  value: BigNumber
+  blockNumber: number
+  addressVotes?: BigNumber
+  totalGroupVotes?: BigNumber
+}
+
+interface ValidatorReward {
+  name: string
+  validator: string
+  validatorPayment: BigNumber
+  validatorScore: BigNumber
+  group: string
+  blockNumber: number
+}
+
+interface ValidatorGroupReward {
+  name: string
+  group: string
+  validator: string
+  groupPayment: BigNumber
+  validatorScore: BigNumber
+  blockNumber: number
+}
+
+export default class Show extends BaseCommand {
+  static description =
+    'Show rewards information about a voter, registered Validator, or Validator Group'
+
+  static flags = {
+    ...BaseCommand.flags,
+    address: Flags.address({ description: 'Address to filter' }),
+    epochs: flags.integer({
+      default: 1,
+      description: 'Show results for the last N epochs',
+    }),
+  }
+
+  static args = []
+
+  static examples = ['show --address 0x5409ed021d9299bf6814279a6a1411a7e866a631']
+
+  async run() {
+    const res = this.parse(Show)
+    const election = await this.kit.contracts.getElection()
+    const validators = await this.kit.contracts.getValidators()
+    const epochSize = await validators.getEpochSize()
+    const currentBlock = await this.web3.eth.getBlockNumber()
+    const lastEpochBlock = Math.floor(currentBlock / epochSize) * epochSize
+    const fromBlock: number = lastEpochBlock - epochSize * ((res.flags.epochs || 1) - 1)
+    const address = res.flags.address || ''
+    let voterRewards: VoterReward[] = []
+    let validatorRewards: ValidatorReward[] = []
+    let validatorGroupRewards: ValidatorGroupReward[] = []
+
+    if (address) {
+      await newCheckBuilder(this)
+        .isAccount(address)
+        .runChecks()
+    }
+
+    // For each epoch...
+    for (let blockNumber = fromBlock; blockNumber <= lastEpochBlock; blockNumber += epochSize) {
+      // Get the groups that address voted for at this epoch, and those groups' total votes.
+      const addressVotes: { [key: string]: BigNumber } = {}
+      const totalGroupVotes: { [key: string]: BigNumber } = {}
+      if (address) {
+        const voter = await election.getVoter(address, blockNumber)
+        for (const vote of voter.votes) {
+          const group: string = vote.group.toLowerCase()
+          addressVotes[group] = (addressVotes[group] || new BigNumber(0)).plus(vote.active)
+          totalGroupVotes[group] = (totalGroupVotes[group] || new BigNumber(0)).plus(
+            await election.getTotalVotesForGroup(group, blockNumber)
+          )
+        }
+      }
+
+      // voterRewardsEvent applies to address when voterRewardsEvent.group in addressVotes.
+      const epochVoterRewardsEvents = await election.getPastEvents(
```

Comment:
i would hide `getPastEvents()` here, and have a methods like `getPastVotesRewards()` or something.